### PR TITLE
GHA: release management for the firebase C++ SDK

### PR DESCRIPTION
**Pull request by compnerd**
Add some rules to release the content from the build to preserve the content.  Furthermore, begin packaging and deploying a nuget package enabling the future consumption within the Arc build.

### CODE REVIEW

File: .github/workflows/bcny-firebase.yml [hunk 1]
```diff
@@ -52,3 +52,37 @@ jobs:
         with:
           name: firebase-windows-amd64
           path: ${{ github.workspace }}/BuildRoot/Library/firebase
+
+      - run: |
+          echo build_tag=$(date +%Y%m%d.$(date +%-H/6 | bc)) >> ${GITHUB_ENV}
```
Comment (fischman-bcny) [inline .github/workflows/bcny-firebase.yml:57]: lol what's up w/ the inner `$()`?
(why is quarter-day a meaningful unit here?)

File: .github/workflows/bcny-firebase.yml [hunk 2]
```diff
@@ -52,3 +52,37 @@ jobs:
...
+      - name: nuget package
+        run: |
+          $treeish = git log -1 --format=%h
+          nuget pack -Properties DESTDIR=${{ github.workspace }}/BuildRoot/Library/firebase -Version 0.0.0.0-${treeish} ${{ github.workspace }}/SourceCache/firebase-cpp-sdk/firebase.nuspec
+        shell: pwsh
+
+      - name: publish nuget
+        env:
+          GITHUB_TOKEN: ${{ secrets.NUGET_TOKEN }}
+          GITHUB_USERNAME: thebrowsercompany-bot2
```
Comment (fischman-bcny) [inline .github/workflows/bcny-firebase.yml:78]: I guess this isn't something we're hoping to upstream?

File: .github/workflows/bcny-firebase.yml [hunk 3]
```diff
@@ -52,3 +52,37 @@ jobs:
...
+          NUGET_SOURCE_NAME: TheBrowserCompany
+          NUGET_PUBLISH_URL: https://nuget.pkg.github.com/thebrowsercompany/index.json
+        run: |
+          nuget sources list | findstr "%NUGET_SOURCE_NAME%"
+          if errorlevel 0 ( nuget sources remove -name %NUGET_SOURCE_NAME% )
+          nuget sources Add -Name %NUGET_SOURCE_NAME% -Source %NUGET_PUBLISH_URL% -username %GITHUB_USERNAME% -password %GITHUB_TOKEN% -StorePasswordInClearText
```
Comment (fischman-bcny) [inline .github/workflows/bcny-firebase.yml:84]: IDK what "StorePasswordInClearText" means but it sounds...exciting?

File: firebase.nuspec [hunk 4]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0"?>
```
Comment (fischman-bcny) [inline firebase.nuspec:1]: I know absolutely nothing about nuget/nuspec so might be nice to get a review from someone who does.

**Review by fischman-bcny — APPROVED** [review]

File: .github/workflows/bcny-firebase.yml (hunk 1 above)
Comment (compnerd) [reply .github/workflows/bcny-firebase.yml:57]: Quarter day is a meaningful unit only in that it is copy-pasta.  We really just need a counter.  The toolchain builds build occur every 6 hours, so we get quarter days.

File: .github/workflows/bcny-firebase.yml (hunk 2 above)
Comment (compnerd) [reply .github/workflows/bcny-firebase.yml:78]: Nope, I see no reason we would ever upstream the nuget package.  If anything, we might consider switching to the upstream package if we could.

File: .github/workflows/bcny-firebase.yml (hunk 3 above)
Comment (compnerd) [reply .github/workflows/bcny-firebase.yml:84]: Exciting is one word for it.  This is copy-pasta from thebrowsercompany/swiftwinrt's rules.

File: .github/workflows/bcny-firebase.yml (hunk 1 above)
Comment (fischman-bcny) [reply .github/workflows/bcny-firebase.yml:57]: lol IMO just use full hours.minutes.seconds and avoid potential future confusion. But that's just me.

**Pull request closed without merging**